contracts/instruments/RibbonThetaVault.sol

```diff
@@ -312,7 +312,11 @@ contract RibbonThetaVault is DSMath, OptionsVaultStorage {
         require(option != address(0), "!option");
         OtokenInterface otoken = OtokenInterface(option);
         require(otoken.isPut() == isPut, "Option type does not match");
-        require(otoken.underlyingAsset() == underlying, "!asset");
+        require(
+            otoken.underlyingAsset() == underlying,
+            "Wrong underlyingAsset"
+        );
+        require(otoken.collateralAsset() == asset, "Wrong collateralAsset");
         require(otoken.strikeAsset() == USDC, "strikeAsset != USDC"); // we just assume all options use USDC as the strike
 
```

Comment:
as you mentioned, Otokens can be collateralized with any arbitrary asset in the future. in theta vault constructor, instead of `asset = _isPut ? _usdc : _asset;` we should pass supplemental collateralAsset parameter that we set asset to.

Reply:
Agreed, let's keep this comment in mind for the yVault integration